jsk_fetch_robot/jsk_fetch_startup/udev_rules/99-edgetpu-accelerator.rules

```diff
@@ -0,0 +1,2 @@
+SUBSYSTEM=="usb",ATTRS{idVendor}=="1a6e",GROUP="plugdev"
+SUBSYSTEM=="usb",ATTRS{idVendor}=="18d1",GROUP="plugdev"
```

Comment:
just out of curiosity, each EdgeTPU have different `idVendor`?

Reply:
I don't know the reason, but product id and vendor id seem to change.

https://dev.to/hb020/comment/c32g
https://stackoverflow.com/a/56675603

Reply:
Coral TPU USB is quire weird device.
When we connect Coral TPU USB, it is recognized as `Global Unichip Corp`.

```
Bus 002 Device 002: ID 1a6e:089a Global Unichip Corp.
```

But after the first run, the device will be recognized as `Google Inc.`

```
$ roslaunch coral_usb edgetpu_object_detector.launch INPUT_IMAGE:=/usb_cam/image_raw
... logging to /home/knorth55/.ros/log/ba441176-1ac1-11ec-a15f-507b9d9efafc/roslaunch-melodic-p50-22462.log
Checking log directory for disk usage. This may take a while.
Press Ctrl-C to interrupt
WARNING: disk usage in log directory [/home/knorth55/.ros/log] is over 1GB.
It's recommended that you use the 'rosclean' command.

started roslaunch server http://melodic-p50:45329/

SUMMARY
========

PARAMETERS
 * /edgetpu_object_detector/device_id: 0
 * /edgetpu_object_detector/image_transport: raw
 * /edgetpu_object_detector/label_file: /home/knorth55/ro...
 * /edgetpu_object_detector/model_file: /home/knorth55/ro...
 * /rosdistro: melodic
 * /rosversion: 1.14.11

NODES
  /
    edgetpu_object_detector (coral_usb/edgetpu_object_detector.py)

ROS_MASTER_URI=http://localhost:11311

process[edgetpu_object_detector-1]: started with pid [22484]
[INFO] [1632218008.656496]: Using transport raw
^C[edgetpu_object_detector-1] killing on exit
[WARN] [1632218013.660277]: [/edgetpu_object_detector] subscribes topics only with child subscribers. Set '~always_subscribe' as True to have it subscribe always.
shutting down processing monitor...
... shutting down processing monitor complete
done

$ lsusb
Bus 002 Device 003: ID 18d1:9302 Google Inc. 
```

Reply:
```
s[knorth55][melodic-p50][~]
$ lsusb
Bus 002 Device 004: ID 1a6e:089a Global Unichip Corp.
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 003: ID 138a:0090 Validity Sensors, Inc.
Bus 001 Device 002: ID 04ca:7058 Lite-On Technology Corp.
Bus 001 Device 012: ID 8087:0a2b Intel Corp.
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
[knorth55][melodic-p50][~]
$ source ~/ros/coral_ws/devel/setup.bash
[knorth55][melodic-p50][~]
$ roslaunch coral_usb edgetpu_object_detector.launch INPUT_IMAGE:=/usb_cam/image_raw
... logging to /home/knorth55/.ros/log/02bc0b5c-1ac2-11ec-a15f-507b9d9efafc/roslaunch-melodic-p50-24916.log
Checking log directory for disk usage. This may take a while.
Press Ctrl-C to interrupt
WARNING: disk usage in log directory [/home/knorth55/.ros/log] is over 1GB.
It's recommended that you use the 'rosclean' command.

started roslaunch server http://melodic-p50:42651/

SUMMARY
========

PARAMETERS
 * /edgetpu_object_detector/device_id: 0
 * /edgetpu_object_detector/image_transport: raw
 * /edgetpu_object_detector/label_file: /home/knorth55/ro...
 * /edgetpu_object_detector/model_file: /home/knorth55/ro...
 * /rosdistro: melodic
 * /rosversion: 1.14.11

NODES
  /
    edgetpu_object_detector (coral_usb/edgetpu_object_detector.py)

ROS_MASTER_URI=http://localhost:11311

process[edgetpu_object_detector-1]: started with pid [24938]
[INFO] [1632218134.507259]: Using transport raw
^C[edgetpu_object_detector-1] killing on exit
[WARN] [1632218139.512827]: [/edgetpu_object_detector] subscribes topics only with child subscribers. Set '~always_subscribe' as True to have it subscribe always.
lsshutting down processing monitor...
... shutting down processing monitor complete
done
us[knorth55][melodic-p50][~]
$ lsusb
Bus 002 Device 005: ID 18d1:9302 Google Inc.
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 003: ID 138a:0090 Validity Sensors, Inc.
Bus 001 Device 002: ID 04ca:7058 Lite-On Technology Corp.
Bus 001 Device 012: ID 8087:0a2b Intel Corp.
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```

Reply:
BTW, quite interesting product, I found. 
8 or 16 edge TPU can be run in parallel with PCIe connector.
https://iot.asus.com/products/AI-accelerator/AI-Accelerator-PCIe-Card/